File: Boj_py/1931.py
N = int(input())

time = []

for i in range(0, N):
    x, y = map(int, input().split())
    time.append([x, y])

time.sort()
time = sorted(time, key=lambda a: a[1])

end = 0
cnt = 0

for i, j in time:
    if i >= end:
        cnt += 1
        end = j


print(cnt)

File: Boj_kt/greedy/1931.kt
fun main() {
    val n = readln().toInt()
    val times = mutableListOf<Pair<Int, Int>>()
    var end = 0
    var cnt = 0

    repeat(n) {
        val input = readln().split(" ").map { it.toInt() }
        times.add(input[0] to input[1])
    }
    times.sortWith(compareBy { it.first })
    times.sortWith(compareBy { it.second })
    for (i in times) {
        if (i.first >= end) {
            cnt++
            end = i.second
        }
    }
    print(cnt)
}